merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/services/GetSimulationResultsAction.java

```diff
@@ -76,6 +79,11 @@ public Response run(final PlanId planId) throws NoSuchPlanException {
     }
   }
 
+  public Map<String, List<Pair<Duration, SerializedValue>>> getResourceSamples(final SimulationDatasetId datasetId) {
+    if (simulationService instanceof final CachedSimulationService css) return css.getResourceSamples(datasetId);
+    throw new RuntimeException("Unable to retrieve resource samples from uncached simulation service");
```

Comment:
Maybe this is not too graceful but we basically can only return resource samples for simulation results that are **cached**. So an uncached simulation service should not support returning resource samples.

For some peace of mind I made `SimulationService` a `sealed` class that can be only `CachedSimulationService`/`UncachedSimulationService`, so this "contract" is at least codified.

Reply:
Hmm... this does make me wonder if we've designed ourselves into a corner a bit. Why is it that the uncached service cannot implement `getResourceSamples`? In theory, I would expect an uncached service's `getResourceSamples` to run a simulation (since it doesn't have cached results), and use those simulation results to produce a response.

Reply:
>Why is it that the uncached service cannot implement getResourceSamples?

The technical reason here is that `getResourceSamples(SimulationDatasetId simulationDatasetId)` doesn't make sense in an uncached scenario (where a simulation dataset must first exist). The Hasura action `resourceSamples` is accepts a `simulationDatasetId` so we have to disallow an uncached sim. service from trying to return resource samples when provided a simulation dataset ID.

We could maybe make another Hasura action for retrieving samples given a plan ID. This action could lazily run a sim. and return samples. However, we should keep in mind that `UncachedSimulationService` is only used in the `DevAppDriver` – not in production. So I'm not sure it deserves a full Hasura action and the additional complexity/bloat that comes with it.

Reply:
Happy to hop on a call about this. The solution I've settled on feels more like a "best of the all bad options".

Reply:
@mattdailis I talked with @patkenneally and we seem to all be in agreeance that `resourceSamples` should be provided a plan ID. This way the cached vs. uncached backends will both be satisfied. The only catch here is that the cached backend will just need to check if a plan has an associated simulation prior to checking for samples.

Reply:
I've reworked this PR to have `resourceSamples` accept a plan ID. :)